truenas_install/__main__.py

```diff
@@ -389,16 +390,48 @@ def main():
                 if probe_dataset_name not in existing_datasets:
                     dataset_name = probe_dataset_name
                     break
+
     run_command([
         "zfs", "create",
         "-o", "mountpoint=legacy",
         "-o", f"truenas:kernel_version={manifest['kernel_version']}",
         "-o", "zectl:keep=False",
         dataset_name,
     ])
+
+    for entry in TRUENAS_DATASETS:
+        cmd = ["zfs", "create", "-u", "-o", "mountpoint=legacy", "-o", "canmount=noauto"]
+        if "NOSUID" in entry["options"]:
+            cmd.extend(["-o", "setuid=off", "-o", "devices=off"])
+        if "NOEXEC" in entry["options"]:
+            cmd.extend(["-o", "exec=off"])
+        if "NODEV" in entry["options"]:
+            cmd.extend(["-o", "devices=off"])
+        if "NOACL" in entry['options']:
+            cmd.extend(["-o", "acltype=off", "-o", "aclmode=discard"])
+        if "NOATIME" in entry["options"]:
+            cmd.extend(["-o", "atime=off"])
+
+        cmd.append(f"{dataset_name}/{entry['name']}")
+        run_command(cmd)
+
     try:
         with tempfile.TemporaryDirectory() as root:
+            undo = []
+            ds_info = []
             run_command(["mount", "-t", "zfs", dataset_name, root])
+            for entry in TRUENAS_DATASETS:
+                this_ds = entry['name']
+                ds_name = f"{dataset_name}/{this_ds}"
+                ds_path = entry.get("mountpoint") or entry["name"]
+                ds_guid = run_command(["zfs", "list", "-o", "guid", "-H", ds_name])
+
+                mp = os.path.join(root, ds_path[1:] if ds_path.startswith('/') else ds_path)
```

Comment:
Just `ds_path.rstrip("/")`, but why are absolute and relative paths are mixed here?